src/we_internal.c

```diff
@@ -1375,6 +1375,7 @@ int wolfengine_bind(ENGINE *e, const char *id)
 
 #ifdef HAVE_FIPS
     wolfCrypt_SetCb_fips(we_fips_cb);
+    PRIVATE_KEY_UNLOCK();
```

Comment:
Should have unlock and lock calls around each private key operation.

Reply:
Once its unlocked, why would we need to lock and unlock it?

Reply:
Don't unlock it globally.
Only unlock it for the shortest amount of time.